### ISSUE TITLE
Configure what constitutes a "low battery" level

### ISSUE THREAD

**Issue by petrklus** [rgc2000/NutClient-ESXi#4]
My ESXi host can take a good few minutes to shut down and I would like to specify a threshold of, say, 40% battery charge level at which I would like to initiate a shutdown.

I used to have the following settings when my TrueNAS was running on the same UPS and I would like to reproduce them with my NutClient on my now ESXi host:

```
override.battery.charge.low = 40
override.battery.charge.warning = 60
```

How could the above be achieved?

**Comment by rgc2000** [comment]
Hi, everything concerning the UPS thresholds to send the events is configurable on the Nut Server only. This means that all the clients shares the settings set in the Nut Server. You can't have different LOWBATT conditions on your clients. They will all receive this event at the same time. This means that if it is already configured on the TrueNAS side, the NutClient-ESXi will de facto use this threshold.

**Comment by petrklus** [comment]
Thank you @rgc2000!

I will check my nut server.

However, the overrides I've mentioned were set on the NUT client of the TrueNAS, I have not changed the NUT server side.

**Comment by rgc2000** [comment]
You are wrong. override.battery.charge.low is configured in ups.conf in the NUT server where you describe and configure the drivers to manage the UPSes attached to the server. Even if your TrueNAS shows it in the client  configuration form it is not part of the NUT client.
Definitively not a client issue. Nothing I can do.

**Comment by petrklus** [comment]
I understand that I need to do the changes on my NUT server, thank you for
the pointer!

For my better understanding, I've checked my TrueNAS setup and it's running
a UPS service in "slave" mode.
Does that mean that TrueNAS runs another NUT server that effectively
proxies the UPS status and thus allows the extra overrides?

I have tested the system behaviour and the TrueNAS VM turns off at the
prescribed 40% override while the parent host does so only after LB signal
is received.

**Comment by rgc2000** [comment]
This is not a truenas support chat. So you will have to dig by yourself. But understand that usually each nut server is also his own client.

**Comment by petrklus** [comment]
Thank you, makes sense - so out of interest, is there plan/merit of
creating a NutServer package for the ESXi? That would then allow for
intercepting/amending low battery thresholds etc.

**Comment by rgc2000** [comment]
It's not posible to create NutServer package running on ESXi because you hav no access the  USB/serial ports from the hypervisor. The ESXi kernel has very little support for external devices. It's not a linux kernel. Then even if it was possible it would be very painful to configure the server because there are much more parameters and also different UPS drivers. On ESXi parameter files are not allowed. Changes are lost on each reboot. That's why on current Nut Client the configuration is stored in UserVars. Then on each reboot the parameter files are rebuilt from the UserVars. One of my best solutions is to use a Raspberry Pi with Raspbian and Nut Server. You don't need a powerful raspberry. If you have a 1st generation Raspberry Pi, this is more than enough. But the very popular solution is to use an office NAS with Nut support embedded.
